### PR TITLE
Add a model for bagged neural networks

### PR DESCRIPTION
**Pull request by topepo**
Engines in tidymodels/baguette#64

### CODE REVIEW

**Comment by simonpcouch** [comment]
The model spec print method will need a new row in this table to describe this model.🙂

https://github.com/tidymodels/parsnip/blob/3c0a0e27662d6325c4d09f2a2be1864c30557342/R/print.R#L39-L40

File: R/bag_mlp.R [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      engine = engine,
+      user_specified_engine = !missing(engine)
+    )
+  }
+
+# ------------------------------------------------------------------------------
+
+#' @method update bag_mlp
+#' @rdname parsnip_update
+#' @inheritParams mars
```
Comment (simonpcouch) [inline R/bag_mlp.R:49]:
```suggestion
```

I believe the intended `@inheritParams` here would be `mlp`, though those params are documented via other `update` functions already in `?parsnip_update`.

File: R/bag_mlp_nnet.R [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#' Bagged neural networks via nnet
+#'
+#' [baguette::bagger()] creates an collection of neural networks forming an
```
Comment (simonpcouch) [inline R/bag_mlp_nnet.R:3]:
```suggestion
#' [baguette::bagger()] creates a collection of neural networks forming an
```

File: R/print.R [hunk 3]
```diff
@@ -38,6 +38,7 @@ model_descs <- tibble::tribble(
   ~cls,                   ~desc,
   "auto_ml",              "Automatic Machine Learning",
   "bag_mars",             "Bagged MARS",
+  "bag_mlp",              "Bagged Neural Networks",
```
Comment (simonpcouch) [inline R/print.R:41]:
```suggestion
  "bag_mlp",              "Bagged Neural Network",
```

...I think? Following the lead of "Bagged Decision Tree" here.

**Review by simonpcouch — COMMENTED** [review]
Looks good! A few small changes in the PR, and 2 questions:

1) Why is `epochs` a tunable parameter for `mlp` but not for `bag_mlp`?

2) If I mistakenly try to tune `epochs`, I get the parsnip error about guarded arguments:

``` r
# pak::pak("tidymodels/baguette#64")

library(baguette)
#> Loading required package: parsnip

bm <-
  bag_mlp(epochs = tune()) %>%
  set_engine("nnet") %>%
  set_mode("regression")

bm_tune <-
  tune::tune_grid(
    bm,
    mpg ~ .,
    rsample::bootstraps(mtcars, 2),
    grid = 2
  )
#> ! Bootstrap1: preprocessor 1/1, model 1/2: The following arguments cannot be manually modified and were removed: ma...
#> ! Bootstrap1: preprocessor 1/1, model 2/2: The following arguments cannot be manually modified and were removed: ma...
#> ! Bootstrap2: preprocessor 1/1, model 1/2: The following arguments cannot be manually modified and were removed: ma...
#> ! Bootstrap2: preprocessor 1/1, model 2/2: The following arguments cannot be manually modified and were removed: ma...

tune::show_notes(bm_tune)
#> unique notes:
#> ────────────────────────────────────────────────────────────────────────────
#> The following arguments cannot be manually modified and were removed: maxit.
```

<sup>Created on 2022-09-22 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Is this the usual error when `tune()` is mistakenly supplied to a main model org that isn't tunable?

**Comment by topepo** [comment]
The epoch issue was hard to track down but fixed in baguette.

**Pull request merged**